pkg/reconciler/v1alpha1/pipeline/tasks.go

```diff
@@ -0,0 +1,39 @@
+/*
+Copyright 2018 The Knative Authors
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+    http://www.apache.org/licenses/LICENSE-2.0
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package pipeline
+
+import (
+	"fmt"
+
+	"github.com/knative/build-pipeline/pkg/apis/pipeline/v1alpha1"
+	listers "github.com/knative/build-pipeline/pkg/client/listers/pipeline/v1alpha1"
+)
+
+// GetTasks retrieves all Tasks instances which the pipeline p references, using
+// lister l. If it is unable to retrieve an instance of a references Task, it will return
+// an error, otherwise it returnsa map from the name of the Task in the Pipeline to the
+// name of the Task object itself.
+func GetTasks(l listers.TaskLister, p *v1alpha1.Pipeline) (map[string]*v1alpha1.Task, error) {
```

Comment:
In the function definition it seems more like a property on Pipeline object.  Is`GetTasks` definition better defined under Pipeline object?
WDYT?

```
func (p *Pipeline) GetTasks(l *lister.TaskLister) (map[string]*v1alpha1.Task, error) 
```

Reply:
No need to move the code under [this]( https://github.com/knative/build-pipeline/blob/master/pkg/apis/pipeline/v1alpha1/pipeline_types.go) file. Just mentioned that function definition could be updated like `func (p *Pipeline) GetTasks(l *lister.TaskLister) (map[string]*v1alpha1.Task, error) `

Reply:
I think we can add this code to `pipeline_types.go` without effecting the generated code - I'll give it a shot, sounds like a good idea!